### PR TITLE
Documentation review

### PR DESCRIPTION
**Pull request by julianbermudez**
This pull request fixes some grammatical and spelling error.

### CODE REVIEW

File: docs/agent.rst [hunk 1]
```diff
@@ -3,34 +3,85 @@
...
+    Specify the `Maximum Transmission Unit` able to send and receive by UDP (default 512).
+
+``CONFIG_TCP_TRANSPORT_MTU``
+    Specify the `Maximum Transmission Unit` able to send and receive by TCP (default 512).
+
+``CONFIG_TCP_MAX_CONNECTIONS``
+    Specify the maximum number of connections which the *Agent* is able to manage (default 100).
+
+``CONFIG_TCP_MAX_BACKLOG_CONNECTIONS``
+    Specify the maximum number of incoming connections which the *Agent* is able to manage (default 100).
```
Comment (lemunozm) [inline docs/agent.rst]: What's exactly the difference with the previous option?

File: docs/agent.rst [hunk 2]
```diff
@@ -3,34 +3,85 @@
...
+    <profiles>
+      <participant profile_name="default_xrce_participant_profile">
+        <rtps>
+          <name>default_participant</name>
+        </rtps>
+      </participant>
+    </profile>
 
-If the transport used is a reliability transport, the use of a best effort stream over it is equivalent to use a reliable stream over a not reliable transport.
+therefore, the *Client* can use `default_xrce_participant_profile` as ``ref`` in the ``mr_write_create_participant_ref`` function.
```
Comment (lemunozm) [inline docs/agent.rst]: The function is called now: ``uxr_buffer_create_participant_ref``

File: docs/deployment.rst [hunk 3]
```diff
@@ -35,22 +35,22 @@ Configurate the publisher
...
         :align: center
 
-In this state a `configurator client` is connected to the agent `A` with the `client key` that will be used by the future `publisher client` (0xAABBCCDD).
-Once a session is logged in, the `configurator client` creates all necesary entities for the `publisher client`.
+In this state a `configurator client` is connected to the *Agent* `A` with the `client key` that will be used by the future `publisher client` (0xAABBCCDD).
+Once a session is logged in, the `configurator client` creates all necessary entities for the `publisher client`.
 This implies the creation of `participant`, `topic`, `publisher`, and `datawriter` entities.
 These entities have a representation as DDS entities, and can be reached now from the DDS world.
-That implies that a possible `subscriber DDS entity` could be already listening topics if it match with a `publisher DDS entity` through `DDS` world.
+That implies that a possible `subscriber DDS entity` could be already listed topics if it match with a `publisher DDS entity` through `DDS` world.
```
Comment (lemunozm) [inline docs/deployment.rst]:
```suggestion
That implies that a possible `subscriber DDS entity` could already be listening topics if it matches with a `publisher DDS entity` through `DDS` world.
```

**Comment by lemunozm** [comment]
The branch must be merge to develop instead of master

File: docs/agent.rst (hunk 1 above)
Comment (julianbermudez) [reply docs/agent.rst]: The first one refers to connection established with different Client, and the second one (incoming) connection pending to establish. I change it a bit.

File: docs/agent.rst (hunk 2 above)
Comment (julianbermudez) [reply docs/agent.rst]: Done!

File: docs/deployment.rst (hunk 3 above)
Comment (julianbermudez) [reply docs/deployment.rst]: Corrected!

File: docs/agent.rst [hunk 4]
```diff
@@ -3,34 +3,85 @@
...
+For UDP communication: ::
+
+    $ ./MicroXRCEAgent udp <port> [<discovery-port>]
+
+For TCP communication: ::
+
+    $ ./MicroXRCEAgent tcp <port> [<discovery-port>]
 
-    $ ./MicroXRCE-DDSAgent serial <device>
+If the transport used is a reliability transport, the use of a best-effort stream over it is equivalent to use a reliable stream over a not reliable transport.
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/agent.rst]: Corrected!

File: docs/agent.rst [hunk 5]
```diff
@@ -3,34 +3,85 @@
...
+        </rtps>
+      </participant>
+    </profile>
 
-If the transport used is a reliability transport, the use of a best effort stream over it is equivalent to use a reliable stream over a not reliable transport.
+therefore, the *Client* can use `default_xrce_participant_profile` as ``ref`` in the ``mr_write_create_participant_ref`` function.
 
-For running the agent is necessary that the file ``DEFAULT_FASTRTPS_PROFILES.xml`` be located in the folder where the
-agent is located. Default installation place this file along with the agent executable into ``/usr/local/bin``.
+Default installation place this file along with the *Agent* executable into ``/usr/local/bin``.
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/agent.rst]: Added!

File: docs/client.rst [hunk 6]
```diff
@@ -5,18 +5,18 @@ Micro XRCE-DDS Client
...
 All functions needed to set up the *Client* can be found into ``client.h`` header.
 This is the only header you need to include.
 
 Profiles
 --------
 
-The client library follows a profile concept that enables to choose add or remove some features in configuration time.
-This allows to reduce the client library size, if there are features that are not used.
+The *Client* library follows a profile concept that enables to choose, add or remove some features in configuration time.
+This allows to reduce the *Client* library size, if there are features that are not used.
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/client.rst]: Done!

File: docs/deployment.rst [hunk 7]
```diff
@@ -3,20 +3,20 @@
...
-This can be done creating two differents clients.
-One for configurating the entities in the agent, and run possibly once, only for creating the entities at configuration time.
-And other/s, that logs in the same session as the configure client (sharing the entities) and only publishes or subscribes data.
+This can be done creating two differents *Clients*.
+One in charge of configure the entities in the *Agent*, and run possibly once, only for creating the entities at configuration time.
+And other/s that logs in the same session as the configure *Client* (sharing the entities) and only publishes or subscribes data.
 
-This way allows to create easy clients in production only with the purpose of send and receive data.
-Related to it, exists the concept of `profile` that allows to build the client library only with the behavior chosen (only publish or only subscribe, for example).
+This way allows to create easily *Clients* in a real scenario only with the purpose of send and receive data.
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/deployment.rst]: Fixed!

File: docs/deployment.rst [hunk 8]
```diff
@@ -3,20 +3,20 @@
...
+Related to it, the concept of `profile` allows to build the *Client* library only with the behavior chosen (only publish or only subscribe, for example).
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/deployment.rst]: Fixed

File: docs/deployment.rst [hunk 9]
```diff
@@ -35,22 +35,22 @@ Configurate the publisher
         :width: 600 px
         :align: center
 
-In this state a `configurator client` is connected to the agent `A` with the `client key` that will be used by the future `publisher client` (0xAABBCCDD).
-Once a session is logged in, the `configurator client` creates all necesary entities for the `publisher client`.
+In this state a `configurator client` is connected to the *Agent* `A` with the `client key` that will be used by the future `publisher client` (0xAABBCCDD).
+Once a session is logged in, the `configurator client` creates all necessary entities for the `publisher client`.
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/deployment.rst]: Fixed!

File: docs/deployment.rst [hunk 10]
```diff
@@ -35,22 +35,22 @@ Configurate the publisher
...
         :width: 600 px
         :align: center
 
-Then, the `publisher client` is connected to the agent `A`.
-This client logs in session with its client key (0xAABBCCDD).
+Then, the `publisher client` is connected to the *Agent* `A`.
+This *Client* logs in session with its *Client* key (0xAABBCCDD).
 At that moment, it can use all entities created related to this `client key`.
-Because all entities that it uses were created successful by the `configurator client`, the `publisher client` can inmediately publish to `DDS`.
+Because all entities that it used were created successful by the `configurator client`, the `publisher client` can immediately publish to `DDS`.
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/deployment.rst]: Fixed!

File: docs/getting_started.rst [hunk 11]
```diff
@@ -64,10 +64,10 @@ Next, we will create a session that allow us interact with the agent:
         return 1;
     }
 
-The first function ``uxr_init_session`` initializes the ``Session`` structure with the transport and the `Client Key` (the session identifier for an agent).
-The ``uxr_set_topic_callback`` function is for registering the function ``on_topic`` which has been called when the `Client` receive a topic.
-Once the session has been initialized, we can send the first message for login the client in agent side: ``uxr_create_session``.
-This function will try to connect with the agent by ``CONFIG_MAX_SESSION_CONNECTION_ATTEMPTS`` attempts (the value can be configurable at ``client.config``).
+The first function ``uxr_init_session`` initializes the ``session`` structure with the transport and the `Client Key` (the session identifier for an *Agent*).
+The ``uxr_set_topic_callback`` function is for registering the function ``on_topic`` which will be called when the `Client` receive a topic.
+Once the session has been initialized, we can send the first message for login the `Client` in *Agent* side: ``uxr_create_session``.
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/getting_started.rst]: Fixed!

File: docs/agent.rst [hunk 12]
```diff
@@ -3,34 +3,86 @@
...
+    Specify the `Maximum Transmission Unit` able to send and receive by UDP (default 512).
+
+``CONFIG_TCP_TRANSPORT_MTU``
+    Specify the `Maximum Transmission Unit` able to send and receive by TCP (default 512).
+
+``CONFIG_TCP_MAX_CONNECTIONS``
+    Specify the maximum number of connections, the *Agent* is able to manage (default 100).
+
+``CONFIG_TCP_MAX_BACKLOG_CONNECTIONS``
+    Specify the maximum number of incoming connections (peding to establish), the *Agent* is able to manage (default 100).
```
[the remark this answers is not in the document]
Comment (julianbermudez) [reply docs/agent.rst]: Well in TCP transport when a Client try to connect with the server it sends a SYN, and the server must response with another SYN. The backlog connections (pending to establish) is queue of SYN messages by client which the server has to manage, that is, finish the 3-way handshake. I hope it helps you.

**Pull request merged**